pkg/scraper/scraper.go

```diff
@@ -18,46 +18,58 @@ const (
 	Bruneida  = 2
 )
 
-type scraper struct {
-	wg   sync.WaitGroup
-	jobs []*model.Job
-}
+type (
+	Scraper interface {
+		// ScrapeJobs scrapes provider's website for job listings and company information if available.
+		// Returned Job struct contains a Company struct with ProviderCompanyID only.
+		// Company details are returned as a separate argument.
+		ScrapeJobs() ([]*model.Job, error)
+	}
 
-func createScraper() scraper {
-	return scraper{wg: sync.WaitGroup{}, jobs: []*model.Job{}}
-}
+	JobCentreScraper struct {
+		BaseURL     string
+		FetchClient FetchClienter
+	}
+
+	BruneidaScraper struct {
+		BaseURL     string
+		FetchClient FetchClienter
+	}
```

Comment:
seems like we can have one implementation of the Scraper interface, since both BruneidaScraper and JobCentreScraper has the same struct

Maybe we can do like this?

```go
type ScraperInterface interface {
    ScrapeJobs() ([]*model.Job, error)
}

type Scraper struct {
    BaseURL    string
    FetchClient FetchClienter
}

```

Reply:
But the implementation of ScrapeJobs are completely different for each of them, so that might not be possible?

The property of the struct is actually for the unit test to be able to mock them rather than for normal use.

Reply:
ooooh that makes a lot of sense!